onnxruntime/python/tools/tensorrt/perf/benchmark.py

```diff
@@ -1047,43 +1034,37 @@ def run_onnxruntime(args, models):
             #######################################
             if args.running_mode == 'benchmark':
                 logger.info("\n----------------------------- benchmark -------------------------------------")
-
-                # resolve providers to create session
-                if is_standalone(ep): 
-                    providers = ep_to_provider_list[trt]
-                else: 
-                    providers = ep_to_provider_list[ep] 
-
-                options = onnxruntime.SessionOptions()
-                options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
-
-                
-                # create onnxruntime inference session
-                try:
-                    sess, _ = create_session(model_path, providers, options)
-
-                except Exception as e:
-                    logger.error(e)
-                    update_fail_model_map(model_to_fail_ep, name, ep, 'runtime error', e)
-                    continue
-                
+        
                 # memory tracking variables 
                 p = None 
                 mem_usage = None
                 result = None
 
                 # get standalone TensorRT perf
                 if is_standalone(ep) and args.trtexec: 
-                    trtexec = True 
                     try: 
-                        result = run_trt_standalone(args.trtexec, name, model_path, sess.get_inputs(), all_inputs_shape, fp16, args.track_memory)
+                        result = run_trt_standalone(args.trtexec, name, model_path, all_inputs_shape, fp16, args.track_memory)
                     except Exception as e: 
                         logger.error(e)
                         update_fail_model_map(model_to_fail_ep, name, ep, 'runtime error', e)
                         continue
 
                 # inference with onnxruntime ep
                 else:         
+                    # resolve providers to create session
+                    providers = ep_to_provider_list[ep] 
+                    options = onnxruntime.SessionOptions()
+                    options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL
```

Comment:
should we make this configurable?
I think there are cases for EP where they don't want the optimized graph and would prefer to do their own graph optimizations?

Reply:
i'm not seeing how this is resolved?
the optimization level is still hardcoded in line 1057 ?